### PR TITLE
Get hetzner sizes from api

### PR DESCRIPTION
**Pull request by kgroschoff**
**What this PR does / why we need it**:
Get hetzner sizes from api

**Special notes for your reviewer**:
/

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Add endpoint to return hetzner sizes from api
```

### CODE REVIEW

File: api/pkg/api/v1/types.go [hunk 1]
```diff
@@ -213,6 +214,47 @@ type AzureSize struct {
...
+	Cores       int                                `json:"cores"`
+	Memory      float32                            `json:"memory"`
+	Disk        int                                `json:"disk"`
+	Pricings    []hcloud.ServerTypeLocationPricing `json:"pricings"`
+	StorageType hcloud.StorageType                 `json:"storageType"`
+	CPUType     hcloud.CPUType                     `json:"cpuType"`
+}
+
+// StorageType specifies the type of storage.
+type StorageType string
```
Comment (zreigz) [inline api/pkg/api/v1/types.go]: if it's only for Hetzner please rename to HetznerStorageType
It seems to never used

File: api/pkg/api/v1/types.go [hunk 2]
```diff
@@ -213,6 +214,47 @@ type AzureSize struct {
...
+
+const (
+	// StorageTypeLocal is the type for local storage.
+	StorageTypeLocal StorageType = "local"
+	// StorageTypeCeph is the type for remote storage.
+	StorageTypeCeph StorageType = "ceph"
+)
+
+// CPUType specifies the type of the CPU.
+type CPUType string
```
Comment (zreigz) [inline api/pkg/api/v1/types.go]: if it's only for Hetzner please rename to HetznerCPUType
It seems to never used

File: api/pkg/api/v1/types.go [hunk 3]
```diff
@@ -213,6 +214,47 @@ type AzureSize struct {
...
+// HetznerSizeList represents an array of Hetzner sizes.
+// swagger:model HetznerSizeList
+type HetznerSizeList struct {
+	Standard  []HetznerSize `json:"standard"`
+	Dedicated []HetznerSize `json:"dedicated"`
+}
+
+// HetznerSize is the object representing Hetzner sizes.
+// swagger:model HetznerSize
+type HetznerSize struct {
```
Comment (zreigz) [inline api/pkg/api/v1/types.go:225]: The final output contains a lot of useless information. Can you display what you really need?


{"standard":[{"id":1,"name":"cx11","description":"CX11","cores":1,"memory":2,"disk":20,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0040000000","Gross":"0.0047600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"2.4900000000","Gross":"2.9631000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0040000000","Gross":"0.0047600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"2.4900000000","Gross":"2.9631000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0040000000","Gross":"0.0047600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"2.4900000000","Gross":"2.9631000000000000"}}],"storageType":"local","cpuType":"shared"},{"id":2,"name":"cx11-ceph","description":"CX11 Ceph Disk","cores":1,"memory":2,"disk":20,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0040000000","Gross":"0.0047600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"2.4900000000","Gross":"2.9631000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0040000000","Gross":"0.0047600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"2.4900000000","Gross":"2.9631000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0040000000","Gross":"0.0047600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"2.4900000000","Gross":"2.9631000000000000"}}],"storageType":"network","cpuType":"shared"},{"id":3,"name":"cx21","description":"CX21","cores":2,"memory":4,"disk":40,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0080000000","Gross":"0.0095200000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"4.9000000000","Gross":"5.8310000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0080000000","Gross":"0.0095200000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"4.9000000000","Gross":"5.8310000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0080000000","Gross":"0.0095200000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"4.9000000000","Gross":"5.8310000000000000"}}],"storageType":"local","cpuType":"shared"},{"id":4,"name":"cx21-ceph","description":"CX21 Ceph Disk","cores":2,"memory":4,"disk":40,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0080000000","Gross":"0.0095200000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"4.9000000000","Gross":"5.8310000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0080000000","Gross":"0.0095200000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"4.9000000000","Gross":"5.8310000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0080000000","Gross":"0.0095200000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"4.9000000000","Gross":"5.8310000000000000"}}],"storageType":"network","cpuType":"shared"},{"id":5,"name":"cx31","description":"CX31","cores":2,"memory":8,"disk":80,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0140000000","Gross":"0.0166600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"8.9000000000","Gross":"10.5910000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0140000000","Gross":"0.0166600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"8.9000000000","Gross":"10.5910000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0140000000","Gross":"0.0166600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"8.9000000000","Gross":"10.5910000000000000"}}],"storageType":"local","cpuType":"shared"},{"id":6,"name":"cx31-ceph","description":"CX31 Ceph Disk","cores":2,"memory":8,"disk":80,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0140000000","Gross":"0.0166600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"8.9000000000","Gross":"10.5910000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0140000000","Gross":"0.0166600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"8.9000000000","Gross":"10.5910000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0140000000","Gross":"0.0166600000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"8.9000000000","Gross":"10.5910000000000000"}}],"storageType":"network","cpuType":"shared"},{"id":7,"name":"cx41","description":"CX41","cores":4,"memory":16,"disk":160,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0260000000","Gross":"0.0309400000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"15.9000000000","Gross":"18.9210000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0260000000","Gross":"0.0309400000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"15.9000000000","Gross":"18.9210000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0260000000","Gross":"0.0309400000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"15.9000000000","Gross":"18.9210000000000000"}}],"storageType":"local","cpuType":"shared"},{"id":8,"name":"cx41-ceph","description":"CX41 Ceph Disk","cores":4,"memory":16,"disk":160,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0260000000","Gross":"0.0309400000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"15.9000000000","Gross":"18.9210000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0260000000","Gross":"0.0309400000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"15.9000000000","Gross":"18.9210000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0260000000","Gross":"0.0309400000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"15.9000000000","Gross":"18.9210000000000000"}}],"storageType":"network","cpuType":"shared"},{"id":9,"name":"cx51","description":"CX51","cores":8,"memory":32,"disk":240,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0500000000","Gross":"0.0595000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"29.9000000000","Gross":"35.5810000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0500000000","Gross":"0.0595000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"29.9000000000","Gross":"35.5810000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0500000000","Gross":"0.0595000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"29.9000000000","Gross":"35.5810000000000000"}}],"storageType":"local","cpuType":"shared"},{"id":10,"name":"cx51-ceph","description":"CX51 Ceph Disk","cores":8,"memory":32,"disk":240,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0500000000","Gross":"0.0595000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"29.9000000000","Gross":"35.5810000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0500000000","Gross":"0.0595000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"29.9000000000","Gross":"35.5810000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0500000000","Gross":"0.0595000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"29.9000000000","Gross":"35.5810000000000000"}}],"storageType":"network","cpuType":"shared"}],"dedicated":[{"id":11,"name":"ccx11","description":"CCX11 Dedicated CPU","cores":2,"memory":8,"disk":80,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0300000000","Gross":"0.0357000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"19.9000000000","Gross":"23.6810000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0300000000","Gross":"0.0357000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"19.9000000000","Gross":"23.6810000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0300000000","Gross":"0.0357000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"19.9000000000","Gross":"23.6810000000000000"}}],"storageType":"local","cpuType":"dedicated"},{"id":12,"name":"ccx21","description":"CCX21 Dedicated CPU","cores":4,"memory":16,"disk":160,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0600000000","Gross":"0.0714000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"34.9000000000","Gross":"41.5310000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0600000000","Gross":"0.0714000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"34.9000000000","Gross":"41.5310000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.0600000000","Gross":"0.0714000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"34.9000000000","Gross":"41.5310000000000000"}}],"storageType":"local","cpuType":"dedicated"},{"id":13,"name":"ccx31","description":"CCX31 Dedicated CPU","cores":8,"memory":32,"disk":240,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.1200000000","Gross":"0.1428000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"69.9000000000","Gross":"83.1810000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.1200000000","Gross":"0.1428000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"69.9000000000","Gross":"83.1810000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.1200000000","Gross":"0.1428000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"69.9000000000","Gross":"83.1810000000000000"}}],"storageType":"local","cpuType":"dedicated"},{"id":14,"name":"ccx41","description":"CCX41 Dedicated CPU","cores":16,"memory":64,"disk":360,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.2400000000","Gross":"0.2856000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"139.9000000000","Gross":"166.4810000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.2400000000","Gross":"0.2856000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"139.9000000000","Gross":"166.4810000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.2400000000","Gross":"0.2856000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"139.9000000000","Gross":"166.4810000000000000"}}],"storageType":"local","cpuType":"dedicated"},{"id":15,"name":"ccx51","description":"CCX51 Dedicated CPU","cores":32,"memory":128,"disk":540,"pricings":[{"Location":{"ID":0,"Name":"fsn1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.4800000000","Gross":"0.5712000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"269.9000000000","Gross":"321.1810000000000000"}},{"Location":{"ID":0,"Name":"nbg1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.4800000000","Gross":"0.5712000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"269.9000000000","Gross":"321.1810000000000000"}},{"Location":{"ID":0,"Name":"hel1","Description":"","Country":"","City":"","Latitude":0,"Longitude":0},"Hourly":{"Currency":"","VATRate":"","Net":"0.4800000000","Gross":"0.5712000000000000"},"Monthly":{"Currency":"","VATRate":"","Net":"269.9000000000","Gross":"321.1810000000000000"}}],"storageType":"local","cpuType":"dedicated"}]}

**Review by zreigz — CHANGES_REQUESTED** [review]

**Comment by kgroschoff** [comment]
Thanks @zreigz. Output now contains following informations:
````
[...]
{
            "id": 1,
            "name": "cx11",
            "description": "CX11",
            "cores": 1,
            "memory": 2,
            "disk": 20
},
[...]

**Pull request merged**